### ISSUE TITLE
Configuration synchronization will not occur if host.zone is not the same as the directory name in zones.d

### ISSUE THREAD

**Issue by 7af03dec0b** [Icinga/icinga2#5390]
Hi,
I use a distributed configuration with master zone and multiple satellite zones. Each zone has on master icinga its own directory in `zones.d`. If the `zone` attribute in the host definition is not the same as the directory in `zones.d`, so the host is not synchronized to the appropriate satellite, and the host remains defined only on the config-master. Such a host is not even replicated on the second master if the master HA setup is used.

For example: I have three zones defined: `master`, `zoneA`, `zoneB` with corresponding directories in `zones.d`. The host definition
```
object Host "Test host" {
    import "generic-host"
    address = "192.168.1.1"
    zone = "zoneA"
}
```
will be correctly replicated to a `zoneA` only if the definition is located in `zones.d/zoneA`. If any other directory is used, the host remains visible only in the master zone (checked using `icinga2 object list -t Host -n "Test host"`). 

If the `zone` attribute is not used in the host definition, the host is correctly replicated to the zone according to the used directory in `zones.d`

## Expected Behavior
If the `zone` attribute in the host/service definition is filled, it has a higher priority than the directory used in the `zones.d` where the configuration is located. 
I would like to use a dynamically generated zone based on an IP address for example:
```
zone = {{ if ( cidr_match("192.168.0.0/24", host.address) ) {
                  return "zoneA"
           } else if ( cidr_match("192.168.1.0/24", host.address) ) {
                  return "zoneB"
           } else {
                  return "master"
           }
       }}
```
or define the same host in multiple zones (network availability monitoring):
```
ZONES = [ "zoneA", "zoneB" ]

for (zone_name in ZONES) {
  object Host "Test host at " + zone_name use (zone_name) {
   import "generic-host"
   address = "192.168.1.1"
   zone = zone_name
   vars.globalping = true
  }
}
```
but now there is no directory where I could place this definition.
## Your Environment

* Version used (`icinga2 --version`): icinga2 - The Icinga 2 network monitoring daemon (version: v2.6.3)
* Operating System and version: CentOS Linux release 7.3.1611 (Core)
* Config validation (`icinga2 daemon -C`): No error in any case tested except when I used a nonexistent zone

**Comment by dnsmichi** [comment]
I do understand the idea, still I am strongly against modifying the behaviour of the file based cluster config sync. When the file persists in `zones.d/zonename` this is the default origin and represented inside the synced structure on the satellite/client. If one would superseed this with a runtime activated object which makes this a runtime synced object, this will get complicated. First off, you'll loose the option to import a template on the remote end, as the fully compiled object is sent through. Second, you'll have a hard time debugging the origin of such a synced object. 

This also is a dangerous area with changing this, as many users tend to pin a specific object to a local zone (or a different one in the cluster). If that would now be changed and synced differently, an entire cluster design breaks in production.

TL;DR - Runtime object sync should only happen from the REST API, and not from a static configuration file. 

Therefore I'm closing this feature request.
